pilot/pkg/config/kube/gateway/controller.go

```diff
@@ -294,14 +294,6 @@ func NewController(
 					Name:      t.Name,
 					Namespace: t.Namespace,
 				}
-			}), false),
-		outputs.ReferenceGrants.collection.RegisterBatch(pushXds(xdsUpdater,
```

Comment:
This was never needed, as evidenced by the fact its adding the wrong `Name`. 

If a ReferenceGrant changes, things depending on it will change.